src/Sema.zig

```diff
@@ -15833,6 +15833,36 @@ fn zirAsm(
             .ComptimeInt => arg.* = try sema.coerce(block, Type.usize, uncasted_arg, src),
             .ComptimeFloat => arg.* = try sema.coerce(block, Type.f64, uncasted_arg, src),
             else => {
+                const input_op_src: LazySrcLoc = .{
+                    .asm_input_op = .{
+                        .asm_node_offset = extra.data.src_node,
+                        .input_index = @intCast(arg_i),
+                    },
+                };
+                if (uncasted_arg_ty.comptimeOnly(mod)) {
+                    return sema.fail(
+                        block,
+                        input_op_src,
+                        "type '{}' is comptime-only and cannot be used for an asm input operand",
+                        .{uncasted_arg_ty.fmt(mod)},
+                    );
+                }
+                if (!uncasted_arg_ty.hasRuntimeBits(mod)) {
```

Comment:
Suggest using `hasRuntimeBitsIgnoreComptime` here. jacobly and I are skeptical of the existence of the "standard" version of this function, and in this case `...IgnoreComptime` is safe because we have the `comptimeOnly` check earlier, so is better to use regardless since it's faster.